CMakeLists.txt

```diff
@@ -117,8 +132,12 @@ set(IREE_EXTERNAL_HAL_DRIVERS "" CACHE STRING "")
 set(IREE_HAL_EXECUTABLE_LOADER_EXTRA_DEPS "" CACHE STRING "")
 
 option(IREE_HAL_DRIVER_DEFAULTS "Sets the default value for all runtime HAL drivers" ON)
-# CUDA support must be explicitly enabled.
+# CUDA support is enabled by default if the platform supports the CUDA SDK and
+# not cross compiling.
 set(IREE_HAL_DRIVER_CUDA_DEFAULT OFF)
+if(IREE_CUDA_SDK_DOWNLOAD_AVAILABLE AND NOT CMAKE_CROSSCOMPILING)
+  set(IREE_HAL_DRIVER_CUDA_DEFAULT ON)
```

Comment:
Doesn't this actually require a GPU with Cuda, not just the stuff we download?

Reply:
Running successfully requires a GPU with compatible drivers, yes. Same for Vulkan. Just building the code does not require drivers though. For tests we have `IREE_VULKAN_DISABLE` and `IREE_CUDA_DISABLE` environment variables and tags to filter on.

Reply:
Vulkan can use swiftshader though :-) I really don't want to end up in a TF situation where disabling Cuda (or at least its warnings) is effectively impossible
. IREE_CUDA_DISABLE + filter tags on the tests I think should do it, but I'd like for us to be sure

Reply:
I think enabling the CUDA build by default is important, and the main thing blocking that was licensing / getting the header files downloaded and configured. Agreed about what we want to avoid. What do you think would help there? More builds that disable certain components?

Reply:
Maybe let's just double check that turning CUDA off is still reasonable here. Adding more builds would be nice, but of course those have a cost

Reply:
Oh wait, our test_all workflow doesn't have a GPU, so it must work :-)

Reply:
Yep! Do you still have concerns with this PR?

Reply:
Nope. Just one typo fix. Thanks for cleaning this up :-)